### DISCUSSION TITLE
“Resource not accessible by integration” when requesting GitHub Pages build via REST API

### DISCUSSION THREAD

**Discussion by btouellette** [community/community#24701] [Apps, API and Webhooks]
<p>I’m attempting to set up a GitHub App that checks in a new resource file and triggers a GitHub Pages build so that the new resource is available via the site. I’m trying to trigger a Pages build manually (as it does not trigger directly on check-in by the GitHub App) with the GitHup App authorization via this endpoint: <a href="https://developer.github.com/v3/repos/pages/#request-a-page-build" rel="nofollow noopener">https://developer.github.com/v3/repos/pages/#request-a-page-build</a></p>
<p>I have the file check in working fine with an installation token so the authorization is solid but I am getting <strong>“Resource not accessible by integration”</strong> back when I attempt to request a new page build.</p>
<p>The GitHub App has access to the repository with permissions ( <strong>Write</strong> access to code, <strong>Read </strong> and <strong>write</strong>  access to pages) and it appears that the pages build API endpoint is enabled for use by GitHub Apps so I’m not clear on why it is returning this error.</p>
<p>This is all being done with <span class="mention">@octokit</span>/rest (<a href="https://www.npmjs.com/package/@octokit/rest" rel="nofollow noopener">https://www.npmjs.com/package/@octokit/rest</a>). Using its debug logging I have verified that the URL/method and the accept header are correct.</p>
<pre><code>method: 'post',
url: 'https://api.github.com/repos/btouellette/HHHFreshBotRedux/pages/builds',
headers: {
    accept: 'application/vnd.github.mister-fantastic-preview+json',
    authorization: 'token v1.snip',
    'user-agent': 'octokit.js/15.15.1 Node.js/10.12.0 (Linux 4.9; x64)'
}
</code></pre>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/btouellette">@btouellette</a>,</p>
<p>Thank you for being here! It seems like you’re authenticating using a GitHub App installation access token. It isn’t currently possible to request page builds as a GitHub App installation (server-to-server request).</p>
<p>While this endpoint is enabled for GitHub Apps, you’ll need to make a user-to-server request to request a page build:</p>
<p><a href="https://developer.github.com/apps/building-github-apps/identifying-and-authorizing-users-for-github-apps/#user-to-server-requests" rel="nofollow noopener">https://developer.github.com/apps/building-github-apps/identifying-and-authorizing-users-for-github-apps/#user-to-server-requests</a></p>
<p>Could you give that a try and let us know if that helps?</p>
<p>Best,</p>
<p>Your GitHub Community Team</p>

**Comment by btouellette** [comment]
<p>Thanks for the info on user-to-server vs server-to-server. This is running as a daily task on a server. I assume user-to-server requests will involve per-session user action (approving access via OAuth URL) so it doesn’t look like I’ll be able to use the page build endpoint.</p>
<p>EDIT: I now have the page build endpoint working by using ‘basic’ authentication in octokit where I can explicitly specify username/pass for my primary account so the page build is requested using the user account credentials rather than the installation token.</p>

**Comment by lancerutkin** [comment]
<p>Hi <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a>,</p>
<p>I think I’m having a similar issue, but would love some clarification. I’m trying to create a fork of a repository with installation token access. I’m getting the same error message as listed above.</p>
<p>Is creating a fork another request that must be done user-to-server?</p>
<p>Lance</p>

**Comment by bryantson** [comment]
<p>Hello. I encountered a similar problem. Try to generate a PAT with right permissions and make sure to authenticate (Settings &gt; Developer settings &gt; PAT &gt; And authenticate for that token). Use that token for your app.</p>
